filebeat/input/v2/input-cursor/manager_test.go

```diff
@@ -0,0 +1,607 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cursor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	pubtest "github.com/elastic/beats/v7/libbeat/publisher/testing"
+	"github.com/elastic/beats/v7/libbeat/tests/resources"
+	"github.com/elastic/go-concert/unison"
+)
+
+type fakeTestInput struct {
+	OnTest func(Source, input.TestContext) error
+	OnRun  func(input.Context, Source, Cursor, Publisher) error
+}
+
+type stringSource string
+
+func TestManager_Init(t *testing.T) {
+	// Integration style tests for the InputManager and the state garbage collector
+
+	t.Run("stopping the taskgroup kills internal go-routines", func(t *testing.T) {
+		numRoutines := runtime.NumGoroutine()
+
+		var grp unison.TaskGroup
+		store := createSampleStore(t, nil)
+		manager := &InputManager{
+			Logger:              logp.NewLogger("test"),
+			StateStore:          store,
+			Type:                "test",
+			DefaultCleanTimeout: 10 * time.Millisecond,
+		}
+
+		err := manager.Init(&grp, v2.ModeRun)
+		require.NoError(t, err)
+
+		time.Sleep(200 * time.Millisecond)
+		grp.Stop()
+
+		// wait for all go-routines to be gone
+
+		for numRoutines < runtime.NumGoroutine() {
```

Comment:
How can we tell if this test fails? Does it run infinitely?

Reply:
`go test` has a default timeout of 10min and fails the complete test run with a stacktrace. If this test is still active after 10min, we got a problem.